nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,536 @@
+          description = ''
+            Database name. When using SQLite this must be an absolute
+            path to the database file.
+          '';
+        };
+
+        user = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = "Database user.";
+        };
+
+        password = mkOption {
+          type = types.str;
+          default = "";
+          description = ''
+            The password corresponding to <option>database.user</option>.
+            Warning: this is stored in cleartext in the Nix store!
+            Use <option>database.passwordFile</option> instead.
+          '';
+        };
+
+        passwordFile = mkOption {
+          type = types.nullOr types.path;
+          default = null;
+          example = "/run/keys/sympa-dbpassword";
+          description = ''
+            A file containing the password corresponding to
+            <option>database.user</option>.
+          '';
+        };
+      };
+
+      web = {
+        enable = mkOption {
+          type = types.bool;
+          default = true;
+          description = "Whether to enable Sympa web interface.";
+        };
+
+        server = mkOption {
+          type = types.enum [ "nginx" "none" ];
+          default = "nginx";
+          description = ''
+            The webserver used for the Sympa web interface. Set it to `none` if you want to configure it yourself.
+            Further nginx configuration can be done by adapting <literal>services.nginx.virtualHosts.&lt;name&gt;</literal>.
+          '';
+        };
+
+        https = mkOption {
+          type = types.bool;
+          default = true;
+          description = ''
+            Whether to use HTTPS. When nginx integration is enabled, this option forces SSL and enables ACME.
+            Please note that Sympa web interface always uses https links even when this option is disabled.
+          '';
+        };
+
+        fcgiProcs = mkOption {
+          type = types.ints.positive;
+          default = 2;
+          description = "Number of FastCGI processes to fork.";
+        };
+      };
+
+      mta = {
+        type = mkOption {
+          type = types.enum [ "postfix" "none" ];
+          default = "postfix";
+          description = ''
+            Mail transfer agent (MTA) integration. Use `none` if you want to configure it yourself.
+
+            The `postfix` integration sets up local Postfix instance that will pass incoming
+            messages from configured domains to Sympa. You still need to configure at least
+            outgoing message handling using e.g. <option>services.postfix.relayHost</option>.
+          '';
+        };
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = "
+          Extra lines to be added verbatim to the main configuration file.
+        ";
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable (mkMerge [
```

Comment:
Why the mkMerge? You could also do `config = mkIf cfg.enable {`

Reply:
I guess it was meant to logically divide the configuration generation based on what options are used. Removed the `mkMerge`.